calculation/seistech_calc/seistech_calc/plots/plt_plotting.py

```diff
@@ -549,3 +592,403 @@ def plot_uhs_branches(
+        color="red",
+        linestyle="solid",
+        label="GCIM - Median",
+        linewidth=1,
+    )
+    plt.plot(
+        periods_list,
+        upper_percen_values,
+        color="red",
+        linestyle="dashdot",
+        label="GCIM - $16^{th}$ Percentile",
+        linewidth=1,
+    )
+
+    for i in range(0, num_gms):
+        plt.plot(
+            periods_list,
+            selected_gms_y_coords[i],
+            color="black",
+            linestyle="solid",
+            label="Selected Ground Motions" if i == 0 else None,
+            linewidth=1,
+        )
+        plt.plot(
+            periods_list,
+            realisations_y_coords[i],
+            color="blue",
+            linestyle="solid",
+            label="Realisations" if i == 0 else None,
+            linewidth=1,
+        )
+
+    # TODO: Investigate log scale issue
+    # plt.xscale("log")
+    plt.yscale("log")
+    plt.xlabel("Period, T (s)")
+    plt.ylabel("Spectral acceleration, SA (g)")
+    plt.title("Pseudo acceleration response spectra")
+    plt.legend()
+
+    if save_file is not None:
+        plt.savefig(save_file)
+        plt.close()
+    else:
+        plt.show()
+
+
+def plot_gms_disagg_distribution(
+    contribution: List,
+    distribution: List,
+    gms_metadata: List,
+    bounds: Dict,
+    metadata: str,
+    save_file: Path = None,
+):
+    """GMS disaggregation distribution plot
+    (Magnitude and Rupture distance)
+
+    Parameters
+    ----------
+    contribution: List
+    distribution: List
+    gms_metadata: List
+    bounds: Dict
+    metadata: str
+    save_file: Path, optional
+    """
+    range_x, range_y = utils.calculate_gms_disagg_distribution(gms_metadata[metadata])
+    bounds_y_range = [0, 1]
+
+    plt.figure(figsize=(16, 9))
+
+    plt.plot(
+        range_x,
+        range_y,
+        color="black",
+        label=const.GMSDisaggDistributionsLabel[metadata].value,
+    )
+    plt.plot(
+        distribution, contribution, label="Disaggregation distribution", color="red"
+    )
+
+    plt.plot(
+        [bounds[metadata]["min"], bounds[metadata]["min"]],
+        bounds_y_range,
+        color="red",
+        linestyle="dotted",
+        label="Lower and upper bound limits",
+    )
+    plt.plot(
+        [bounds[metadata]["max"], bounds[metadata]["max"]],
+        bounds_y_range,
+        color="red",
+        linestyle="dotted",
+    )
+
+    if metadata == "rrup":
+        plt.xscale("log")
+    plt.xlabel(const.GMSDisaggDistributionsLabel[metadata].value)
+    plt.ylabel("Cumulative Probability, CDF")
+    plt.ylim(ymin=0)
+    plt.title(f"{const.GMSDisaggDistributionsLabel[metadata].value} distribution")
+    plt.legend()
+
+    if save_file is not None:
+        plt.savefig(save_file)
+        plt.close()
+    else:
+        plt.show()
+
+
+def plot_gms_available_gm(
+    metadata: Dict,
+    bounds: Dict,
+    num_in_bounds: int,
+    save_file: Path = None,
+):
+    """GMS Available ground motions plot
```

Comment:
Can we extend the docstrings for these  functions a bit more, i.e. be more specific in terms of what plot is produced, e.g.
`Plot of GMs in the GM dataset with respect to distance (Rrup) and Magnitude (Mw). Also shows the causal parameters used during ground motion selection`

Reply:
If you are not sure about what to put, post a picture of the plot along with the function name on the slack channel and I can help you with an appropriate description.